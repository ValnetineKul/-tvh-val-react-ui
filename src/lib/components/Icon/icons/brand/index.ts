export { ReactComponent as Facebook } from './svg/facebook.svg';
export { ReactComponent as Instagram } from './svg/instagram.svg';
export { ReactComponent as LinkedIn } from './svg/linkedin.svg';
export { ReactComponent as Twitter } from './svg/twitter.svg';
export { ReactComponent as Whatsapp } from './svg/whatsapp.svg';
export { ReactComponent as Youtube } from './svg/youtube.svg';
