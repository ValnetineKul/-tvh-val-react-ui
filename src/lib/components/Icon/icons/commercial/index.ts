export { ReactComponent as QuoteLeftSolid } from './svg/quoteLeftSolid.svg';
export { ReactComponent as StarHalfSolid } from './svg/starHalfSolid.svg';
export { ReactComponent as StarSolid } from './svg/starSolid.svg';
