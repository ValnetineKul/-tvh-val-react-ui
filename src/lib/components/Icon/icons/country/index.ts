export { ReactComponent as Australia } from './svg/australia.svg';
export { ReactComponent as Belarus } from './svg/belarus.svg';
export { ReactComponent as Belgium } from './svg/belgium.svg';
export { ReactComponent as China } from './svg/china.svg';
export { ReactComponent as Cyprus } from './svg/cyprus.svg';
export { ReactComponent as Denmark } from './svg/denmark.svg';
export { ReactComponent as England } from './svg/england.svg';
export { ReactComponent as France } from './svg/france.svg';
export { ReactComponent as Germany } from './svg/germany.svg';
export { ReactComponent as India } from './svg/india.svg';
export { ReactComponent as Mexico } from './svg/mexico.svg';
export { ReactComponent as Russia } from './svg/russia.svg';
export { ReactComponent as UnitedKingdom } from './svg/unitedKingdom.svg';
export { ReactComponent as Usa } from './svg/usa.svg';
