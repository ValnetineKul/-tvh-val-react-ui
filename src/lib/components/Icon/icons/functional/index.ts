export { ReactComponent as AddressBook } from './svg/addressBook.svg';
export { ReactComponent as Analytics } from './svg/analytics.svg';
export { ReactComponent as AngleDown } from './svg/angleDown.svg';
export { ReactComponent as AngleLeft } from './svg/angleLeft.svg';
export { ReactComponent as AngleRight } from './svg/angleRight.svg';
export { ReactComponent as AngleUp } from './svg/angleUp.svg';
export { ReactComponent as ArrowAltCircleLeft } from './svg/arrowAltCircleLeft.svg';
export { ReactComponent as ArrowDown } from './svg/arrowDown.svg';
export { ReactComponent as ArrowDownArrowUp } from './svg/arrowDownArrowUp.svg';
export { ReactComponent as ArrowLeft } from './svg/arrowLeft.svg';
export { ReactComponent as ArrowLeftToLine } from './svg/arrowLeftToLine.svg';
export { ReactComponent as ArrowRight } from './svg/arrowRight.svg';
export { ReactComponent as ArrowRightArrowLeft } from './svg/arrowRightArrowLeft.svg';
export { ReactComponent as ArrowRightToLine } from './svg/arrowRightToLine.svg';
export { ReactComponent as ArrowUp } from './svg/arrowUp.svg';
export { ReactComponent as ArrowUpRightFromSquare } from './svg/arrowUpRightFromSquare.svg';
export { ReactComponent as Atlas } from './svg/atlas.svg';
export { ReactComponent as Award } from './svg/award.svg';
export { ReactComponent as Badge } from './svg/badge.svg';
export { ReactComponent as BadgePercent } from './svg/badgePercent.svg';
export { ReactComponent as BallotCheck } from './svg/ballotCheck.svg';
export { ReactComponent as Bars } from './svg/bars.svg';
export { ReactComponent as BatteryFull } from './svg/batteryFull.svg';
export { ReactComponent as BatteryHalf } from './svg/batteryHalf.svg';
export { ReactComponent as BatteryLow } from './svg/batteryLow.svg';
export { ReactComponent as Bell } from './svg/bell.svg';
export { ReactComponent as Book } from './svg/book.svg';
export { ReactComponent as BookAlt } from './svg/bookAlt.svg';
export { ReactComponent as BookmarkCheck } from './svg/bookmarkCheck.svg';
export { ReactComponent as BookOpen } from './svg/bookOpen.svg';
export { ReactComponent as Box } from './svg/box.svg';
export { ReactComponent as BoxAlt } from './svg/boxAlt.svg';
export { ReactComponent as Boxes } from './svg/boxes.svg';
export { ReactComponent as BoxFull } from './svg/boxFull.svg';
export { ReactComponent as BoxHeart } from './svg/boxHeart.svg';
export { ReactComponent as BoxOpen } from './svg/boxOpen.svg';
export { ReactComponent as Briefcase } from './svg/briefcase.svg';
export { ReactComponent as Building } from './svg/building.svg';
export { ReactComponent as BulkyItem } from './svg/bulkyItem.svg';
export { ReactComponent as BullseyeArrow } from './svg/bullseyeArrow.svg';
export { ReactComponent as BullseyePointer } from './svg/bullseyePointer.svg';
export { ReactComponent as Calculator } from './svg/calculator.svg';
export { ReactComponent as Calendar } from './svg/calendar.svg';
export { ReactComponent as CalendarCheck } from './svg/calendarCheck.svg';
export { ReactComponent as CalendarStar } from './svg/calendarStar.svg';
export { ReactComponent as Camera } from './svg/camera.svg';
export { ReactComponent as CarMechanic } from './svg/carMechanic.svg';
export { ReactComponent as CartArrowDown } from './svg/cartArrowDown.svg';
export { ReactComponent as Cat } from './svg/cat.svg';
export { ReactComponent as ChartLine } from './svg/chartLine.svg';
export { ReactComponent as ChartPie } from './svg/chartPie.svg';
export { ReactComponent as Check } from './svg/check.svg';
export { ReactComponent as CheckCircle } from './svg/checkCircle.svg';
export { ReactComponent as ChessKnight } from './svg/chessKnight.svg';
export { ReactComponent as ClipboardListCheck } from './svg/clipboardListCheck.svg';
export { ReactComponent as Clock } from './svg/clock.svg';
export { ReactComponent as Cog } from './svg/cog.svg';
export { ReactComponent as Cogs } from './svg/cogs.svg';
export { ReactComponent as Coin } from './svg/coin.svg';
export { ReactComponent as Coins } from './svg/coins.svg';
export { ReactComponent as Comment } from './svg/comment.svg';
export { ReactComponent as CommentsAlt } from './svg/commentsAlt.svg';
export { ReactComponent as ContainerStorage } from './svg/containerStorage.svg';
export { ReactComponent as ConveyorBelt } from './svg/conveyorBelt.svg';
export { ReactComponent as ConveyorBeltSlash } from './svg/conveyorBeltSlash.svg';
export { ReactComponent as Copy } from './svg/copy.svg';
export { ReactComponent as CreditCard } from './svg/creditCard.svg';
export { ReactComponent as DesktopAlt } from './svg/desktopAlt.svg';
export { ReactComponent as Dolly } from './svg/dolly.svg';
export { ReactComponent as DollyFlatbed } from './svg/dollyFlatbed.svg';
export { ReactComponent as DollyFlatbedAlt } from './svg/dollyFlatbedAlt.svg';
export { ReactComponent as Download } from './svg/download.svg';
export { ReactComponent as EllipsisH } from './svg/ellipsisH.svg';
export { ReactComponent as Envelope } from './svg/envelope.svg';
export { ReactComponent as ExclamationCircle } from './svg/exclamationCircle.svg';
export { ReactComponent as ExclamationTriangle } from './svg/exclamationTriangle.svg';
export { ReactComponent as Eye } from './svg/eye.svg';
export { ReactComponent as EyeSlash } from './svg/eyeSlash.svg';
export { ReactComponent as Fast } from './svg/fast.svg';
export { ReactComponent as FileAlt } from './svg/fileAlt.svg';
export { ReactComponent as FileCheck } from './svg/fileCheck.svg';
export { ReactComponent as FileDownload } from './svg/fileDownload.svg';
export { ReactComponent as FileInvoice } from './svg/fileInvoice.svg';
export { ReactComponent as FileUpload } from './svg/fileUpload.svg';
export { ReactComponent as Filter } from './svg/filter.svg';
export { ReactComponent as FlagAlt } from './svg/flagAlt.svg';
export { ReactComponent as Folder } from './svg/folder.svg';
export { ReactComponent as Forklift } from './svg/forklift.svg';
export { ReactComponent as Fragile } from './svg/fragile.svg';
export { ReactComponent as Gauge } from './svg/gauge.svg';
export { ReactComponent as GaugeMax } from './svg/gaugeMax.svg';
export { ReactComponent as GaugeMin } from './svg/gaugeMin.svg';
export { ReactComponent as Gift } from './svg/gift.svg';
export { ReactComponent as GiftCard } from './svg/giftCard.svg';
export { ReactComponent as Gifts } from './svg/gifts.svg';
export { ReactComponent as Globe } from './svg/globe.svg';
export { ReactComponent as GlobeAfrica } from './svg/globeAfrica.svg';
export { ReactComponent as GlobeAmericas } from './svg/globeAmericas.svg';
export { ReactComponent as GlobeAsia } from './svg/globeAsia.svg';
export { ReactComponent as GlobeEurope } from './svg/globeEurope.svg';
export { ReactComponent as GraduationCap } from './svg/graduationCap.svg';
export { ReactComponent as Grid } from './svg/grid.svg';
export { ReactComponent as HandHoldingBox } from './svg/handHoldingBox.svg';
export { ReactComponent as HandPointerSwipe } from './svg/handPointerSwipe.svg';
export { ReactComponent as Handshake } from './svg/handshake.svg';
export { ReactComponent as Headset } from './svg/headset.svg';
export { ReactComponent as HelmetSafety } from './svg/helmetSafety.svg';
export { ReactComponent as Hiking } from './svg/hiking.svg';
export { ReactComponent as History } from './svg/history.svg';
export { ReactComponent as Home } from './svg/home.svg';
export { ReactComponent as HorizontalRule } from './svg/horizontalRule.svg';
export { ReactComponent as Image } from './svg/image.svg';
export { ReactComponent as Inbox } from './svg/inbox.svg';
export { ReactComponent as Indent } from './svg/indent.svg';
export { ReactComponent as Infinite } from './svg/infinite.svg';
export { ReactComponent as InfoCircle } from './svg/infoCircle.svg';
export { ReactComponent as Inventory } from './svg/inventory.svg';
export { ReactComponent as InvoiceOverdue } from './svg/invoiceOverdue.svg';
export { ReactComponent as Language } from './svg/language.svg';
export { ReactComponent as Leaf } from './svg/leaf.svg';
export { ReactComponent as LevelRight } from './svg/levelRight.svg';
export { ReactComponent as LightbulbOn } from './svg/lightbulbOn.svg';
export { ReactComponent as Link } from './svg/link.svg';
export { ReactComponent as LinkSlash } from './svg/linkSlash.svg';
export { ReactComponent as List } from './svg/list.svg';
export { ReactComponent as Lock } from './svg/lock.svg';
export { ReactComponent as LockOpen } from './svg/lockOpen.svg';
export { ReactComponent as MapLocationDot } from './svg/mapLocationDot.svg';
export { ReactComponent as MapMarkerAlt } from './svg/mapMarkerAlt.svg';
export { ReactComponent as Medal } from './svg/medal.svg';
export { ReactComponent as Megaphone } from './svg/megaphone.svg';
export { ReactComponent as Microchip } from './svg/microchip.svg';
export { ReactComponent as Minus } from './svg/minus.svg';
export { ReactComponent as MobileButton } from './svg/mobileButton.svg';
export { ReactComponent as MoneyBillWave } from './svg/moneyBillWave.svg';
export { ReactComponent as MoneyCheckEdit } from './svg/moneyCheckEdit.svg';
export { ReactComponent as Narwhal } from './svg/narwhal.svg';
export { ReactComponent as NonReturnableItem } from './svg/nonReturnableItem.svg';
export { ReactComponent as Outdent } from './svg/outdent.svg';
export { ReactComponent as Pallet } from './svg/pallet.svg';
export { ReactComponent as Paperclip } from './svg/paperclip.svg';
export { ReactComponent as PaperPlane } from './svg/paperPlane.svg';
export { ReactComponent as ParachuteBox } from './svg/parachuteBox.svg';
export { ReactComponent as Pen } from './svg/pen.svg';
export { ReactComponent as PeopleArrowsLeftRight } from './svg/peopleArrowsLeftRight.svg';
export { ReactComponent as PeopleGroup } from './svg/peopleGroup.svg';
export { ReactComponent as PersonDolly } from './svg/personDolly.svg';
export { ReactComponent as PersonRays } from './svg/personRays.svg';
export { ReactComponent as Phone } from './svg/phone.svg';
export { ReactComponent as Plane } from './svg/plane.svg';
export { ReactComponent as PlaneSlash } from './svg/planeSlash.svg';
export { ReactComponent as Play } from './svg/play.svg';
export { ReactComponent as PlayCircle } from './svg/playCircle.svg';
export { ReactComponent as Plus } from './svg/plus.svg';
export { ReactComponent as Print } from './svg/print.svg';
export { ReactComponent as PuzzlePiece } from './svg/puzzlePiece.svg';
export { ReactComponent as QrCode } from './svg/qrCode.svg';
export { ReactComponent as QuestionCircle } from './svg/questionCircle.svg';
export { ReactComponent as Random } from './svg/random.svg';
export { ReactComponent as Recycle } from './svg/recycle.svg';
export { ReactComponent as Redo } from './svg/redo.svg';
export { ReactComponent as ResearchRequest } from './svg/researchRequest.svg';
export { ReactComponent as ReturnableItem } from './svg/returnableItem.svg';
export { ReactComponent as Ribbon } from './svg/ribbon.svg';
export { ReactComponent as RocketLaunch } from './svg/rocketLaunch.svg';
export { ReactComponent as Route } from './svg/route.svg';
export { ReactComponent as RulerHorizontal } from './svg/rulerHorizontal.svg';
export { ReactComponent as Scanner } from './svg/scanner.svg';
export { ReactComponent as Screwdriver } from './svg/screwdriver.svg';
export { ReactComponent as Search } from './svg/search.svg';
export { ReactComponent as SearchMinus } from './svg/searchMinus.svg';
export { ReactComponent as SearchPlus } from './svg/searchPlus.svg';
export { ReactComponent as ShieldCheck } from './svg/shieldCheck.svg';
export { ReactComponent as Ship } from './svg/ship.svg';
export { ReactComponent as ShippingFast } from './svg/shippingFast.svg';
export { ReactComponent as ShippingTimed } from './svg/shippingTimed.svg';
export { ReactComponent as ShoppingBasket } from './svg/shoppingBasket.svg';
export { ReactComponent as ShoppingCart } from './svg/shoppingCart.svg';
export { ReactComponent as SignOut } from './svg/signOut.svg';
export { ReactComponent as Smile } from './svg/smile.svg';
export { ReactComponent as Snowplow } from './svg/snowplow.svg';
export { ReactComponent as SolarPanel } from './svg/solarPanel.svg';
export { ReactComponent as Sparkles } from './svg/sparkles.svg';
export { ReactComponent as Star } from './svg/star.svg';
export { ReactComponent as Stopwatch } from './svg/stopwatch.svg';
export { ReactComponent as Sync } from './svg/sync.svg';
export { ReactComponent as Tag } from './svg/tag.svg';
export { ReactComponent as Tags } from './svg/tags.svg';
export { ReactComponent as ThumbsUp } from './svg/thumbsUp.svg';
export { ReactComponent as Ticket } from './svg/ticket.svg';
export { ReactComponent as TimelineArrow } from './svg/timelineArrow.svg';
export { ReactComponent as Times } from './svg/times.svg';
export { ReactComponent as TimesCircle } from './svg/timesCircle.svg';
export { ReactComponent as Tools } from './svg/tools.svg';
export { ReactComponent as Tractor } from './svg/tractor.svg';
export { ReactComponent as Trash } from './svg/trash.svg';
export { ReactComponent as Trophy } from './svg/trophy.svg';
export { ReactComponent as TrophyAlt } from './svg/trophyAlt.svg';
export { ReactComponent as Truck } from './svg/truck.svg';
export { ReactComponent as TruckContainer } from './svg/truckContainer.svg';
export { ReactComponent as TruckFree } from './svg/truckFree.svg';
export { ReactComponent as TruckFreeSlash } from './svg/truckFreeSlash.svg';
export { ReactComponent as TruckLoading } from './svg/truckLoading.svg';
export { ReactComponent as TruckPickup } from './svg/truckPickup.svg';
export { ReactComponent as TruckPlow } from './svg/truckPlow.svg';
export { ReactComponent as UnknownPart } from './svg/unknownPart.svg';
export { ReactComponent as Unlink } from './svg/unlink.svg';
export { ReactComponent as UpRightAndDownLeftFromCenter } from './svg/upRightAndDownLeftFromCenter.svg';
export { ReactComponent as User } from './svg/user.svg';
export { ReactComponent as UserCircle } from './svg/userCircle.svg';
export { ReactComponent as UserHeadset } from './svg/userHeadset.svg';
export { ReactComponent as Users } from './svg/users.svg';
export { ReactComponent as UserShield } from './svg/userShield.svg';
export { ReactComponent as Video } from './svg/video.svg';
export { ReactComponent as VideoSlash } from './svg/videoSlash.svg';
export { ReactComponent as View360 } from './svg/view360.svg';
export { ReactComponent as Wallet } from './svg/wallet.svg';
export { ReactComponent as WalletCheck } from './svg/walletCheck.svg';
export { ReactComponent as WarehouseFull } from './svg/warehouseFull.svg';
export { ReactComponent as WeightHanging } from './svg/weightHanging.svg';
export { ReactComponent as Wheat } from './svg/wheat.svg';
export { ReactComponent as Wifi } from './svg/wifi.svg';
export { ReactComponent as WifiSlash } from './svg/wifiSlash.svg';
export { ReactComponent as Wrench } from './svg/wrench.svg';
