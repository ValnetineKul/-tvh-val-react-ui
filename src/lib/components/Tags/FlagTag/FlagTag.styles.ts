import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles<void, 'patchLabel' | 'extraPatchLabel' | 'labelText' | 'tagBox' | 'icon' | 'textIcon'>({
  name: 'FlagTag',
})((theme, _params, classes) => ({
  root: {
    [`& .${classes.patchLabel}`]: {
      backgroundColor: theme.color['bg/surface200/default'],
      color: theme.color['text/onFeedback'],
      '&:after': {
        backgroundColor: theme.color['bg/surface200/default'],
      },
    },
    [`& .${classes.extraPatchLabel}`]: {
      backgroundColor: theme.color['bg/surface300/default'],
      color: theme.color['text/onFeedback'],
      '&:before, &:after': {
        backgroundColor: `${theme.color['bg/surface300/default']} !important`,
      },
    },
  },
  labelText: {},
  tagBox: {
    display: 'inline-flex',
    paddingRight: theme.layoutSpacing['spacing/300'],
    overflow: 'hidden',
    [`& .${classes.labelText}`]: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: `${theme.layoutSpacing['spacing/200']} ${theme.layoutSpacing['spacing/300']}`,
    },
  },
  patchLabel: {
    position: 'relative',
    ...theme.font['screen/tag/commercial2'],
    '&:after': {
      content: '""',
      position: 'absolute',
      top: 0,
      right: -2,
      width: 10,
      height: '100%',
      transform: 'skew(-10deg)',
    },
  },
  extraPatchLabel: {
    marginLeft: theme.layoutSpacing['spacing/200'],
    '&:before': {
      content: '""',
      position: 'absolute',
      top: 0,
      width: 10,
      height: '100%',
      marginLeft: -2, // because of the skewed borders
      transform: 'skew(-10deg)',
    },
  },
  icon: {
    position: 'relative', // fix for storybook iframe
    zIndex: 10,
  },
  noLabel: {
    [`& .${classes.tagBox}`]: {
      paddingRight: theme.layoutSpacing['spacing/300'],
    },
    [`& .${classes.extraPatchLabel}`]: {
      marginLeft: theme.layoutSpacing['spacing/000'],
      '&:before': {
        backgroundColor: 'transparent !important',
        width: 0,
        height: 0,
      },
    },
  },
  sizeXs: {
    [`& .${classes.icon}`]: {
      margin: `${theme.layoutSpacing['spacing/200']}`,
    },
  },
  sizeSm: {
    [`& .${classes.patchLabel}`]: {
      ...theme.font['screen/tag/commercial1'],
      '&:after': {
        right: -3,
      },
    },
    [`& .${classes.patchLabel} + .${classes.patchLabel}`]: {
      marginLeft: 6,
    },
    [`& .${classes.extraPatchLabel}`]: {
      '&:before': {
        marginLeft: -3,
      },
    },
    [`& .${classes.icon}`]: {
      margin: `${theme.layoutSpacing['spacing/200']}`,
    },
  },
  sizeMd: {
    [`& .${classes.patchLabel}`]: {
      ...theme.font['screen/tag/commercial1'],
      '&:after': {
        right: -4,
      },
    },
    [`& .${classes.patchLabel} + .${classes.patchLabel}`]: {
      marginLeft: 7,
    },
    [`& .${classes.extraPatchLabel}`]: {
      '&:before': {
        marginLeft: -4,
      },
    },
    [`& .${classes.icon}`]: {
      margin: theme.layoutSpacing['spacing/300'],
    },
    [`& .${classes.labelText}`]: {
      padding: theme.layoutSpacing['spacing/300'],
    },
  },
  textIconWrapper: {
    display: 'flex',
    alignItems: 'center',
  },
  textIcon: {
    padding: `0 ${theme.layoutSpacing['spacing/100']}`,
  },
}));

export default useStyles;
