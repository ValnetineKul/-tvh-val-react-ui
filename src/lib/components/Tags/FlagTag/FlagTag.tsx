import type { FC } from 'react';
import React from 'react';
import { capitalize } from '@mui/material/utils';
import Icon from '../../Icon';
import Typography from '../../Typography';
import useStyles from './FlagTag.styles';

export interface FlagTagProps {
  icon:
    | React.FunctionComponent<
        React.SVGProps<SVGSVGElement> & {
          title?: string;
        }
      >
    | string;
  size?: 'xs' | 'sm' | 'md';
  classNames?: { root?: string; patch?: string; extraPatch?: string };
  message: string;
  hasLabel?: boolean;
}

const FlagTag: FC<FlagTagProps> = ({ message, size = 'md', classNames, hasLabel = true, icon }) => {
  const { classes, cx } = useStyles();

  const sizeClassName = (classes as Record<string, string>)[`size${capitalize(size)}`];
  const rootClassName = cx(classes.root, sizeClassName, classNames && classNames.root, {
    [classes.noLabel]: !hasLabel,
  });
  const patchLabelClassName = cx(
    classes.patchLabel,
    classes.extraPatchLabel,
    classNames && classNames.patch,
    classNames && classNames.extraPatch,
    { [classes.textIconWrapper]: typeof icon === 'string' }
  );

  const printLabel = () => {
    return hasLabel ? (
      <Typography
        variant="inherit"
        className={cx(classes.patchLabel, classes.labelText, classNames && classNames.patch)}
      >
        {message}
      </Typography>
    ) : null;
  };

  const printIcon = () => {
    return typeof icon === 'string' ? (
      <span className={cx(classes.icon, classes.textIcon)}>{icon}</span>
    ) : (
      <Icon icon={icon} size={size === 'xs' ? 'sm' : 'md'} aria-label={message} className={classes.icon} />
    );
  };

  return (
    <div className={rootClassName}>
      <div className={classes.tagBox}>
        {printLabel()}
        <div className={patchLabelClassName}>{printIcon()}</div>
      </div>
    </div>
  );
};

export default FlagTag;
