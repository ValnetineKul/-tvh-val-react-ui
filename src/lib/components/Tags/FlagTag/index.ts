import FlagTag from './FlagTag';

export type { FlagTagProps } from './FlagTag';
export default FlagTag;
