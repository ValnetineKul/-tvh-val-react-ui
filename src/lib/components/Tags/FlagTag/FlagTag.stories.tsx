import { createMeta, createTemplate } from '../../../story-utils';
import FlagTag from './FlagTag';
import { Sparkles } from '../../Icon/icons/functional';

const Template = createTemplate(FlagTag);

export const Primitive = Template.bind({});

Primitive.args = {
  icon: Sparkles,
  size: 'md',
  message: 'label',
  hasLabel: true,
};

Primitive.storyName = 'FlagTag';

export default createMeta({
  component: FlagTag,
  title: 'Components/Tags/FlagTag',
  argTypes: {
    icon: {
      control: {
        disable: true,
      },
    },
  },
});
