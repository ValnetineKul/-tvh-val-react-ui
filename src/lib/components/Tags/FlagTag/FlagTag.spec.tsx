import React from 'react';
import { render, screen } from '../../../test-utils';
import { Sparkles } from '../../Icon/icons/functional';
import FlagTag from './FlagTag';

describe('FlagTag', () => {
  it('Should render correctly', () => {
    const { container } = render(<FlagTag message="promo" icon={Sparkles} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should not render label', () => {
    render(<FlagTag message="promo" icon={Sparkles} hasLabel={false} />);
    expect(screen.queryByText(/PROMO/)).not.toBeInTheDocument();
  });

  it('Should render text icon', () => {
    render(<FlagTag message="warranty" icon="2Y" hasLabel={false} />);
    expect(screen.getByText(/2Y/)).toBeInTheDocument();
  });

  describe('Should set size', () => {
    const cases = [
      [undefined, 'FlagTag-sizeMd'],
      ['xs', 'FlagTag-sizeXs'],
      ['sm', 'FlagTag-sizeSm'],
      ['md', 'FlagTag-sizeMd'],
    ];
    it.each(cases)('given %p, returns %p', (input: React.ComponentProps<typeof FlagTag>['size'], expected: string) => {
      const { container } = render(<FlagTag message="promo" icon={Sparkles} size={input} />);
      expect((container.firstChild as HTMLElement).className).toMatch(expected);
    });
  });
});
