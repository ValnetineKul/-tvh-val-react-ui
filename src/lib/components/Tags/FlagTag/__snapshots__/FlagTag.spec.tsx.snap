// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagTag Should render correctly 1`] = `
<div
  class="tss-2beij6-FlagTag-root-FlagTag-sizeMd"
>
  <div
    class="tss-75p4hn-FlagTag-tagBox tss-2-FlagTag-tagBox-ref"
  >
    <span
      class="tss-2-FlagTag-patchLabel-ref tss-2-FlagTag-labelText-ref tss-fczuv6-Typography-root-FlagTag-patchLabel-FlagTag-labelText"
    >
      promo
    </span>
    <div
      class="tss-2-FlagTag-patchLabel-ref tss-2-FlagTag-extraPatchLabel-ref tss-1trdz71-FlagTag-patchLabel-FlagTag-extraPatchLabel"
    >
      <div
        class="tss-2-FlagTag-icon-ref tss-1lgqc9e-Icon-root-Icon-sizeMd-FlagTag-icon"
      >
        <svg
          aria-hidden="true"
          aria-label="promo"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
          color="inherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          sparkles.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
